### PR TITLE
Added F410T(8-B)Y board

### PR DESCRIPTION
**Pull request by alextrical**
Added F410T(8-B)Y board

Testing how to use Branches for development work

### CODE REVIEW

**Review by ABOSTM — COMMENTED** [review]
Just to be sure we are aligned, when I said 1 PR per series, I mean 1 for all STM43F4 MCU.
But I understand that you are experimenting.
Otherwise, this PR LGTM :-)

File: README.md [hunk 1]
```diff
@@ -252,6 +252,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | :green_heart:  | STM32F407ZE<br>STM32F407ZG | Generic Board | **2.0.0** |  |
 | :green_heart:  | STM32F410C8<br>STM32F410CB | Generic Board | *1.9.0* |  |
 | :green_heart:  | STM32F410R8<br>STM32F410RB | Generic Board | *1.9.0* |  |
+| :yellow_heart: | STM32F410T8<br>STM32F410TB | Generic Board | **2.3.0** |  |
```
Comment (fpistm) [inline README.md:255]:
```suggestion
| :yellow_heart: | STM32F410T8<br>STM32F410TB | Generic Board | **2.4.0** |  |
```

File: variants/STM32F4xx/F410T(8-B)Y/generic_clock.c [hunk 2]
```diff
@@ -20,8 +20,45 @@
...
+  RCC_OscInitStruct.HSIState = RCC_HSI_ON;
+  RCC_OscInitStruct.HSICalibrationValue = RCC_HSICALIBRATION_DEFAULT;
+  RCC_OscInitStruct.PLL.PLLState = RCC_PLL_ON;
+  RCC_OscInitStruct.PLL.PLLSource = RCC_PLLSOURCE_HSI;
+  RCC_OscInitStruct.PLL.PLLM = 8;
+  RCC_OscInitStruct.PLL.PLLN = 100;
+  RCC_OscInitStruct.PLL.PLLP = RCC_PLLP_DIV2;
+  RCC_OscInitStruct.PLL.PLLQ = 4;
+  RCC_OscInitStruct.PLL.PLLR = 2;
+  if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK)
```
Comment (fpistm) [inline variants/STM32F4xx/F410T(8-B)Y/generic_clock.c:44]: Astyle should be applied here

**Review by fpistm — CHANGES_REQUESTED** [review]

File: variants/STM32F4xx/F410T(8-B)Y/generic_clock.c (hunk 2 above)
Comment (alextrical) [reply variants/STM32F4xx/F410T(8-B)Y/generic_clock.c]: I applied the style to the temp folders storing the files for all the Variants prior to merging. In not sure why it didn't apply. I will take note to double check for future PR's

**Comment by alextrical** [comment]
> Just to be sure we are aligned, when I said 1 PR per series, I mean 1 for all STM43F4 MCU.

Correct, I just started using the smallest group I had ;)

**Review by ABOSTM — APPROVED** [review]

File: variants/STM32F4xx/F410T(8-B)Y/ldscript.ld [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+_estack = ORIGIN(RAM) + LENGTH(RAM); /* end of "RAM" Ram type memory */
+
+_Min_Heap_Size = 0x200 ; /* required amount of heap */
+_Min_Stack_Size = 0x400 ; /* required amount of stack */
+
+/* Memories definition */
+MEMORY
+{
+  RAM    xrw    : ORIGIN = 0x20000000,   LENGTH = LD_MAX_DATA_SIZE
+  FLASH    rx    : ORIGIN = 0x8000000 + LD_FLASH_OFFSET, LENGTH = LD_MAX_SIZE - LD_FLASH_OFFSET
```
Comment (ABOSTM) [inline variants/STM32F4xx/F410T(8-B)Y/ldscript.ld:48]: CI check fails (compilation) due to missing parenthesis
```suggestion
  RAM    (xrw)    : ORIGIN = 0x20000000,   LENGTH = LD_MAX_DATA_SIZE
  FLASH    (rx)    : ORIGIN = 0x8000000 + LD_FLASH_OFFSET, LENGTH = LD_MAX_SIZE - LD_FLASH_OFFSET
```

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: variants/STM32F4xx/F410T(8-B)Y/ldscript.ld (hunk 3 above)
Comment (alextrical) [reply variants/STM32F4xx/F410T(8-B)Y/ldscript.ld]: Fixed issues caused by a broken RegEx replacementt

**Review by ABOSTM — APPROVED** [review]
LGTM

**Comment by fpistm** [comment]
Hi @alextrical 
Thanks for the update anyway, a merge is not a rebase and provides artifact which should not appears in a PR.
Due to the merge you pull also several commit already available in the main.

**Review by fpistm — APPROVED** [review]
LGTM

**Pull request merged**